spec/index.html

```diff
@@ -8421,6 +8421,165 @@ <h4>SPARQL Query</h4>
             one query level in the tree.</p>
         </section>
       </section>
+      <section id="algebraicSyntax">
+        <h3>Algebraic Syntax</h3>
+        <p>To define the evaluation semantics of SPARQL query strings
+          these strings are first translated into a more abstract, algebraic syntax.
```

Comment:
The process is that the grammar defines an AST (Abstract Syntax Tree) which is translated into an expression in the SPARQL algebra. (Maybe it show also say something about once translated, it should be evaluated to get the answers.)

The strings are not translated directly.
It is a specific algebra - the SPARQL algebra.

```suggestion
        <p>The evaluation semantics of a SPARQL query is defined by translating the
         SPARQL query string into an abstract syntax tree of the
         <a href="#sparqlGrammar">SPARQL grammar</a>
         then into an expression of the SPARQL Algebra.
```

I prefer the singular form that emphases the one-to-one translation. You may prefer the pluralized style.

Reply:
> I prefer the singular form that emphases the one-to-one translation

I think I agree ... but it's hard to be sure without a parallel expression of the pluralized style to compare.

Reply:
"more abstract" seems odd which is why I suggested explciitly naming the SPARQL Algebra (the only choice of algebra).

Reply:
That's OK - thanks.

Just suggestions:

"evaluation semantics of a SPARQL query" -> "evaluation semantics of a SPARQL query,"

"into a syntax" -> into a structure"

Reply:
Adding the suggested comma into my previous proposal:

```suggestion
        <p>To define the evaluation semantics of a SPARQL query,
          the abstract syntax tree of the SPARQL query string
          (as defined by the <a href="#sparqlGrammar">SPARQL grammar</a>)
          is first translated into a syntax that resembles the
          <a href="#sparqlAlgebra">SPARQL algebra</a>.
```